### PR TITLE
Update runtime-debug error handling

### PR DESCRIPTION
**Pull request by MDrakos**
<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-2124" title="DX-2124" target="_blank">TaskDX-2124</a>  Categorize runtime-debug errors
  </td></table>
  <br />

### CODE REVIEW

**Review by Naatan — CHANGES_REQUESTED** [review]

File: pkg/platform/api/buildplanner/model/buildplan.go [hunk 1]
```diff
@@ -111,6 +112,10 @@ func (e *BuildPlannerError) UserError() string {
 }
 
 func (e *BuildPlannerError) Error() string {
+	if e.Err != nil {
+		return e.Err.Error()
+	}
```
[the remark this answers is not in the document]
Comment (MDrakos) [reply pkg/platform/api/buildplanner/model/buildplan.go:117]: This change also introduces the `Err` field: https://github.com/ActiveState/cli/pull/2737/files#diff-14612946db10f18744a68890b9e353cf87d4da9104592d9fe4346264ed83ef77R95

[the remark this answers is not in the document]
Comment (Naatan) [reply pkg/platform/api/buildplanner/model/buildplan.go:117]: I get that, but that doesn't do anything to let us identify the error type, which is what this story is about. I'm not against also including this change, I just want to understand what this change does? What type of scenario are we not handling prior to this change?

**Review by Naatan — APPROVED** [review]
LGTM. I don't have any issue with the change I'm still commenting on other than wanting to understand what error conditions this would raise.

File: pkg/platform/api/buildplanner/model/buildplan.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (MDrakos) [reply pkg/platform/api/buildplanner/model/buildplan.go:117]: We are already identifying build planner errors but this has only been solving errors. Adding the error field and returning it first allows us to also capture other errors that we want to be considered as build planner errors. In this PR I've included deprecation and fallback errors when processing a build planner response to this type so they will also be identified as build planner errors.

**Pull request merged**